Add existence check for runtime_path

saschagrunert opened the pull request:
Hey there :wave:, during my experiments with the `Runtimes` configuration variable I noticed that there is no check for existence of the `runtime_path`. This pull request contains an additional sanity check as well as a debug output to see which runtimes got registered during server instantiation.

For later improvements I would suggest to move the configuration validation into the `config.go` files for a better test-ability. Would you agree to that? :innocent:

saschagrunert commented:
/assign @rhatdan 
/assign @cyphar

giuseppe commented:
/test all

giuseppe commented:
/ok-to-test

giuseppe commented:
/test all

saschagrunert commented:
It seems that `runc` is not available on the travis machine, which we now need for a proper execution. Do we want to add it as dependency there?

mrunalp commented:
I think this check makes more sense at runtime rather than config generation. A config could be generated anywhere and then copied to the nodes where crio will run so the machine where it is generated shouldn't require the runtime.

rhatdan commented:
/test all
/approve

rhatdan commented:
LGTM

mrunalp commented:
This looks fine but the commits need to be cleaned up. The first commit adds code that is subsequently moved in the second commit.

saschagrunert commented:
> This looks fine but the commits need to be cleaned up. The first commit adds code that is subsequently moved in the second commit.

@mrunalp You're right, I rebased on top of the latest master and squashed them together.

mrunalp commented:
/test all

mrunalp commented:
/test e2e_rhel

mrunalp commented:
lgtm
